django/db/models/sql/compiler.py

```diff
@@ -709,7 +709,7 @@ def find_ordering_name(self, name, opts, alias=None, default_order='ASC',
             already_seen = already_seen or set()
             join_tuple = tuple(getattr(self.query.alias_map[j], 'join_cols', None) for j in joins)
             if join_tuple in already_seen:
-                raise FieldError('Infinite loop caused by ordering.')
+                raise FieldError("Infinite loop caused by ordering (\"%s\" on %s)." % (join_tuple, field))
```

Comment:
What's the point of switching to outer double quotes if you have to escape them? Stick to single quote and use `%r` instead of double quote wrapping

```suggestion
                raise FieldError('Infinite loop caused by ordering (%r on %s).' % (join_tuple, field))
```